lib/SyTest/Identity/Server.pm

```diff
@@ -0,0 +1,121 @@
+package SyTest::Identity::Server;
+
+use strict;
+use warnings;
+
+use base qw( Net::Async::HTTP::Server );
+
+use Crypt::NaCl::Sodium;
+use List::Util qw( any );
+use Protocol::Matrix qw( encode_base64_unpadded sign_json );
+use SyTest::HTTPServer::Request;
+use HTTP::Response;
+
+my $crypto_sign = Crypt::NaCl::Sodium->sign;
+
+sub _init
+{
+   my $self = shift;
+   my ( $params ) = @_;
+
+   $self->rotate_keys;
+
+   $self->{bindings} = {};
+   $self->{expected_tokens} = {};
+
+   # Use 'on_request' as a configured parameter rather than a subclass method
+   # so that the '$CLIENT_LOG' logic in run-tests.pl can properly put
+   # debug-printing wrapping logic around it.
+   $params->{on_request} = \&on_request;
+   $params->{request_class} ||= "SyTest::HTTPServer::Request";
+
+   return $self->SUPER::_init( @_ );
+}
+
+sub rotate_keys
+{
+   my $self = shift;
+
+   ( $self->{public_key}, $self->{private_key} ) = $crypto_sign->keypair;
+
+   $self->{keys} = {
+      "ed25519:0" => encode_base64_unpadded( $self->{public_key} ),
+   };
+
+}
+
+sub on_request
+{
+   my $self = shift;
+   my ( $req ) = @_;
+
+   my $path = $req->path;
+   my %resp;
+
+   if( $path eq "/_matrix/identity/api/v1/pubkey/isvalid" ) {
+      my $user_agent = $req->header( "User-Agent" );
+      if( defined $self->{isvalid_needs_useragent} and !$user_agent =~ m/\Q$self->{isvalid_needs_useragent}/ ) {
+         die "Wrong useragent made /isvalid request";
+      }
+      $resp{valid} = (any { $_ eq $req->query_param("public_key") } values $self->{keys}) ?
+         JSON::true : JSON::false;
+      $req->respond_json( \%resp );
+   } elsif( my ( $key_name ) = $path =~ m#^/_matrix/identity/api/v1/pubkey/([^/]*)$# ) {
```

Comment:
Uncuddle the `elsif` and `else` keywords; they want to start a new line after the previous block's close brace.